torch/_functorch/partitioners.py

```diff
@@ -1921,6 +1921,43 @@ def estimate_for_budget(b):
     )[0]
 
 
+def _broadcast_rank0_decision(joint_graph, saved_values):
+    # use the same policy across different GPUs
+    from torch._subclasses.fake_tensor import unset_fake_temporarily
+
+    def has_collectives(joint_graph):
+        for node in joint_graph.nodes:
+            if isinstance(
+                node.target, torch._ops.OpOverload
+            ) and node.target.namespace in {"_c10d_functional", "c10d_functional"}:
+                return True
+        return False
+
+    def has_same_args(joint_graph):
+        inputs = [x.name for x in joint_graph.find_nodes(op="placeholder")]
+        all_inputs = [None for _ in range(torch.distributed.get_world_size())]
+        with no_dispatch(), unset_fake_temporarily():
+            torch.distributed.all_gather_object(all_inputs, inputs)
```

Comment:
can we either:

(1) tweak this collective to just compute a hash on the graph (without symint metas) and check that the hashes are the same

(2) if not, add a comment calling out the failure mode here and the potential followup?